jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ErrorHandler.java

```diff
@@ -106,7 +108,8 @@ public boolean handle(Request request, Response response, Callback callback) thr
         {
             try
             {
-                errorDispatcher.error(servletContextRequest.getHttpServletRequest(), servletContextRequest.getHttpServletResponse());
+                contextHandler.requestInitialized(servletContextRequest, httpServletRequest);
```

Comment:
We can't call `requestInitialized`/`requestDestroyed` for errors from the `ServletChannel` because we don't know if we will dispatch to an error page or not. 

So we avoid calling it from the `SEND_ERROR` state and defer it to the `ErrorHandler` to decide to call it. 

There should be no way for `requestInitialized` to be previously called here, as we specifically didn't call it in `ServletChannel.dispatch`.

Reply:
Perhaps a comment her referencing the javadoc that says we call listeners before/after entering/exiting first filter/servlet of a dispatch?